@here/harp-mapview/lib/VisibleTileSet.ts

```diff
@@ -404,6 +404,12 @@ export class VisibleTileSet {
                 maxElevation = Math.max(maxElevation, tile.maxElevation);
             });
         });
+        // We process buildings elevation yet (via ElevationRangeSource or either way), so correct
+        // the highest elevation to account for highest possible building.
+        maxElevation =
+            maxElevation < EarthConstants.MAX_ELEVATION - EarthConstants.MAX_BUILDING_HEIGHT
+                ? maxElevation + EarthConstants.MAX_BUILDING_HEIGHT
+                : maxElevation;
```

Comment:
Shouldn't it be:`: EarthConstants.MAX_ELEVATION` ?

Reply:
Nope, we want to take into account all buildings possible that otherwise could be clipped out by near plane. Expression is created this way, that it always preserve EarthConstants.MAX_ELEVATION as absolute maximum, either way we already know that there is no buildings on Mt.Everest or any 8000er :).
Further on there is idea to sample tiles geometry and provide actual buildings data (maximum height on the tile), but this is not a part of that topic/PR/issue. Here we need to take terrain into account, yet without breaking with buildings rendering.